app/src/main/java/org/kiwix/kiwixmobile/KiwixMobileActivity.java

```diff
@@ -920,13 +924,53 @@ public void sendContactEmail() {
   @Override
   public void openExternalUrl(Intent intent) {
     if (intent.resolveActivity(getPackageManager()) != null) {
-      startActivity(intent);
+      // Show popup with warning that this url is external and could lead to additional costs
+      // or may event not work when the user is offline.
+      if (intent.hasExtra("external_link")
+          && intent.getBooleanExtra("external_link", false)
+          && isExternalLinkPopup) {
+        externalLinkPopup(intent);
+      } else {
+        startActivity(intent);
+      }
     } else {
       String error = getString(R.string.no_reader_application_installed);
       Toast.makeText(this, error, Toast.LENGTH_LONG).show();
     }
   }
 
+  private void externalLinkPopup(Intent intent) {
+    new AlertDialog.Builder(this, dialogStyle())
+        .setTitle(R.string.external_link_popup_dialog_title)
+        .setMessage(R.string.external_link_popup_dialog_message)
+        .setNegativeButton(android.R.string.no, new DialogInterface.OnClickListener() {
+          @Override
+          public void onClick(DialogInterface dialogInterface, int i) {
+            // do nothing
+          }
+        })
+        .setNeutralButton(R.string.do_not_ask_anymore, new DialogInterface.OnClickListener() {
+          @Override
+          public void onClick(DialogInterface dialogInterface, int i) {
+            PreferenceManager
+                .getDefaultSharedPreferences(KiwixMobileActivity.this)
+                .edit()
+                .putBoolean(PREF_EXTERNAL_LINK_POPUP, false)
+                .apply();
+            isExternalLinkPopup = false;
+
+            startActivity(intent);
+          }
+        })
+        .setPositiveButton(android.R.string.yes, new DialogInterface.OnClickListener() {
+          @Override
+          public void onClick(DialogInterface dialogInterface, int i) {
+            startActivity(intent);
+          }
+        })
+        .setIcon(android.R.drawable.ic_dialog_alert)
```

Comment:
I personally don't like the alert icon. We might be using it in other places though. Did you see this somewhere else in the app?

Reply:
Yes, it is being used on dialog which pops up in Settings->Clear history

Reply:
Ok, then this is good.